File: lib-android/src/main/java/com/pranav/lib_android/code/disassembler/JavapDisassembler.java
package com.pranav.lib_android.code.disassembler;

import com.sun.tools.javap.JavapTask;

import java.io.StringWriter;
import java.util.ArrayList;

public class JavapDisassembler {

    final String path;

    public JavapDisassembler(String classFile) {
        path = classFile;
    }

    public String disassemble() throws Throwable {
        // Create an arraylist for storing javap arguments
        ArrayList<String> args = new ArrayList<>();
        args.add("-c");
        args.add(classFile);
        // Create a StringWriter object that will store the output
        StringWriter writer = new StringWriter();
        // Create a JavapTask to handle the arguments
        JavapTask task = new JavapTask();
        task.setLog(writer);
        task.run(args.toArray(new String[0]));
        // return the disassembled file as string
        return writer.toString();
    }
}

File: android-compiler/src/main/java/org/cosmic/ide/code/disassembler/JavapDisassembler.kt
package org.cosmic.ide.code.disassembler

import com.sun.tools.javap.JavapTask
import java.io.StringWriter

class JavapDisassembler(
    private val path: String
) {

    @Throws(Throwable::class)
    fun disassemble(): String {
        // Create an arraylist for storing javap arguments
        val args = listOf(
            "-c",
            path
        )
        // Create a StringWriter object that will store the output
        val writer = StringWriter()
        // Create a JavapTask to handle the arguments
        val task = JavapTask()
        task.handleOptions(args.toTypedArray())
        task.setLog(writer)
        task.run()
        // return the disassembled file as string
        return writer.toString()
    }
}
